### PR TITLE
A user can handle its own tokens

### PR DESCRIPTION
**Pull request by teryk**

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

File: server/sonar-server/src/main/java/org/sonar/server/usertoken/ws/GenerateAction.java [hunk 1]
```diff
@@ -63,13 +62,12 @@ public void define(WebService.NewController context) {
       .setPost(true)
       .setDescription("Generate a user access token. <br />" +
         "Please keep your tokens secret. They enable to authenticate and analyze projects.<br />" +
-        "It requires administration permissions.")
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/usertoken/ws/GenerateAction.java]: "permissions" --> singular ?

File: server/sonar-server/src/test/java/org/sonar/server/usertoken/ws/GenerateActionTest.java [hunk 2]
```diff
@@ -93,6 +97,15 @@ public void return_json_example() {
...
+
+    GenerateWsResponse response = newRequest(null, TOKEN_NAME);
+
+    assertThat(response.getLogin()).isEqualTo(GRACE_HOPPER);
+  }
+
+  @Test
   public void fail_if_login_does_not_exist() {
     expectedException.expect(NotFoundException.class);
     expectedException.expectMessage("User with login 'unknown-login' not found");
```
Comment (simonbrandhof) [inline server/sonar-server/src/test/java/org/sonar/server/usertoken/ws/GenerateActionTest.java]: This is a vulnerability issue. If I'm not an administrator, I can know if an arbitrary login exists by calling /api/user_tokens/generate?name=test&login=foo. This request returns:
`{"errors":[{"msg":"User with login 'admin2' not found"}]}` whereas it should return `{"errors":[{"msg":"Insufficient privileges"}]}`

Comment (simonbrandhof) [inline server/sonar-server/src/test/java/org/sonar/server/usertoken/ws/GenerateActionTest.java]: Error message of `POST api/user_tokens/generate?name=test` with anonymous user could be improved:
```
{"errors":[{"msg":"User with login 'null' not found"}]}
```

**Comment by simonbrandhof** [comment]
LGTM

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Pull request closed without merging**